cpp/src/arrow/compute/exec.h

```diff
@@ -442,5 +443,44 @@ Result<Datum> CallFunction(const std::string& func_name, const ExecBatch& batch,
 
 /// @}
 
+/// \defgroup compute-function-executor One-shot calls to obtain function executors
+///
+/// @{
+
+/// \brief One-shot executor provider for all types of functions.
+///
+/// Does kernel dispatch and argument checking, while iteration of ChunkedArray inputs
+/// and wrapping of outputs are deferred to the executor.
+ARROW_EXPORT
+Result<std::shared_ptr<FunctionExecutor>> GetFunctionExecutor(
+    const std::string& func_name, const std::vector<Datum>& args,
```

Comment:
Same here, I would expect a `const std::vector<TypeHolder>&`.

Reply:
Will do. I'll also point to `GetFunctionArgumentTypes` from the doc when the caller has args.

Reply:
I ended up adding an overload here and leaving `GetFunctionArgumentTypes` internal.